pkg/apis/config/v1alpha1/resourceinterpretercustomization_types.go

```diff
@@ -40,23 +42,195 @@ type ResourceInterpreterCustomizationSpec struct {
 	// and not try to change it.
 	// +optional
 	Retention LocalValueRetention `json:"retention,omitempty"`
+
+	// ReplicaResource describes the rules for Karmada to discover the resource's
+	// replica as well as resource requirements.
+	// It would be useful for those CRD resources that declare workload types like
+	// Deployment.
+	// It is usually not needed for Kubernetes native resources(Deployment, Job) as
+	// Karmada knows how to discover info from them. But if it is set, the default
+	// discovery rule will be ignored.
+	// +optional
+	ReplicaResource ReplicaResourceRequirement `json:"replicaResource,omitempty"`
```

Comment:
Just a question. Why not use struct pointer?

Reply:
Right. Good point. optional filed should use pointer.

Reply:
done.